internal/credit/ledger.go

```diff
@@ -117,6 +105,32 @@ func (f LedgerEntryList) GetEntries() []LedgerEntry {
 	return list
 }
 
+func (f LedgerEntryList) Len() int {
+	return len(f.list)
+}
+
+// Truncate removes all entries after the limit.
+func (f LedgerEntryList) Truncate(limit int) LedgerEntryList {
+	if limit >= len(f.list) {
+		return f
```

Comment:
Note that returning the slices allows it to be modified. Not sure if returning a clone is necessary, just something to be mindful of.

Reply:
Fortunately that's not the case here, but thank you for noting this.